src/processor/reconnection-graph.service.ts

```diff
@@ -439,4 +457,38 @@ export class ReconnectionGraphService {
       }
     }
   }
+
+  async waitForProviderWebhookRecovery(providerId: ProviderId | string): Promise<void> {
+    const providerAPI = this.getProviderAPI(providerId);
+    const healthCheckSuccessesThreshold: number = this.configService.getHealthCheckSuccessThreshold();
+    const healthCheckRetryIntervalMilliseconds: number = this.configService.getHealthCheckRetryIntervalSeconds() * 1000;
+    let healthCheckSuccesses: number = 0;
+
+    while (healthCheckSuccesses < healthCheckSuccessesThreshold) {
+      try {
+        await providerAPI.get(`/api/v1.0.0/health`);
+        healthCheckSuccesses++;
+      } catch {
+        // Reset healthCheckSuccesses to 0 on failure. We will not go out of waiting for recovery until there
+        // are a number of sequential healthy responses equaling healthCheckSuccessesThreshold.
+        healthCheckSuccesses = 0;
+      }
+
+      await new Promise(r => setTimeout(r, healthCheckRetryIntervalMilliseconds));
+    }
+  }
+
+  getProviderAPI(providerId: ProviderId | string): AxiosInstance {
+    const headers = {
+      Authorization: 'Bearer <access_token>', // Replace with your actual access token if required
+    };
```

Comment:
Need to resolve this:
Since MeWe has indicated they won't be using any auth token for their endpoint, let's make the access token optional in the config, and only fill it in here if present.

Reply:
This is now optional